main.py

```diff
@@ -290,34 +304,39 @@ def quantize_aq(
                 )
             layer = layer.to(dtype=layer_dtype_original)
             print("FINISHED FINETUNING")
-        if args.save:
+
+        if args.save and not loaded_layer:
             os.makedirs(args.save, exist_ok=True)
             layer_save_path = os.path.join(args.save, f"{layer_index}.pth")
             print(f"Saving layer {layer_index}... to {layer_save_path}")
             torch.save(layer, layer_save_path)
+            if args.on_save:
+                exec(args.on_save)  # a callback e.g. to save progress in slurm or similar distributed infrastructure
 
+        should_compute_mse = not (args.skip_out_loss or loaded_layer)
```

Comment:
Rationale: if we loaded the layer, we do not record pre-quantization outputs to save time; hence, we do not have the references for mse. Doing so would increase the time it takes to resume by at least 2x.